### PR TITLE
Add k1 test functions

### PR DESCRIPTION
**Pull request by jonas-lj**
## Description 

Adds test functions to generate keypairs and create ECDSA signatures over Secp256k1

## Test plan 

Unit tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-framework/packages/sui-framework/sources/crypto/ecdsa_k1.move [hunk 1]
```diff
@@ -49,4 +49,36 @@ module sui::ecdsa_k1 {
...
+    /// @param private_key: A 32-bytes private key that is used to sign the message.
+    /// @param msg: The message to sign, this is raw message without hashing.
+    /// @param hash: The hash function used to hash the message when signing.
+    ///
+    /// Return the 64-bytes signature in form (r, s) that is signed using Secp256k1.
+    /// This should ONLY be used in tests, because it will reveal the private key onchain.
+    public native fun secp256k1_sign(private_key: &vector<u8>, msg: &vector<u8>, hash: u8): vector<u8>;
+
+    #[test_only]
+    #[allow(unused_field)]
```
Comment (dariorussi) [inline crates/sui-framework/packages/sui-framework/sources/crypto/ecdsa_k1.move:63]: do we need this? I assume given the public getters, all fields are used?

File: sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs [hunk 2]
```diff
@@ -286,3 +293,74 @@ pub fn secp256k1_verify(
...
+
+pub fn secp256k1_keypair_from_seed(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 1);
+
+    // This function is only used for testing, so we don't need to charge gas
```
Comment (dariorussi) [inline sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs:342]: is this just for now? I do not understand what it means for a public function to assume it is used only in testing

File: sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs [hunk 3]
```diff
@@ -286,3 +293,74 @@ pub fn secp256k1_verify(
...
+
+pub fn secp256k1_sign(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 3);
+
+    // This function is only used for testing, so we don't need to charge gas
```
Comment (dariorussi) [inline sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs:305]: is this just for now? I do not understand what it means for a public function to assume it is used only in testing

**Review by dariorussi — COMMENTED** [review]
question follows

File: sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs (hunk 2 above)
Comment (jonas-lj) [reply sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs:342]: No, this function should never be used for anything but test purposes. The corresponding move functions are annotated with `#[test_only]` but I don't think we can do anything for the Rust functions besides adding a comment.

Comment (dariorussi) [reply sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs]: you are right this is a rust function, my bad

File: sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs [hunk 4]
```diff
@@ -286,3 +295,98 @@ pub fn secp256k1_verify(
...
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 3);
+
+    // The corresponding Move function, sui::ecdsa_k1::secp256k1_sign, is only used for testing, so
+    // we don't need to charge any gas.
+    let cost = 0.into();
```
Comment (jonas-lj) [inline sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs:316]: I've just hard-coded the zero gas cost here, but we could also get it from `NativesCostTable` if that's better?

Comment (dariorussi) [reply sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs:317]: don't know, I think this is fine? Not sure if @tzakian has got opinion on the specifics here

**Review by dariorussi — APPROVED** [review]
tested on the bridge test framework by adding calls to those functions and it works great!
Thanks.
Maybe let's wait for someone else in the Move team to take a look?
If no one does by tomorrow let's go in

**Review by damirka — APPROVED** [review]
Looks good on the Move end for me. Can't add much about the native function implementation.

File: sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs (hunk 4 above)
Comment (tzakian) [reply sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs:317]: As long as the native is test only it should be fine :) And in fact is better for gas profiling. So LGTM.

**Pull request merged**